dali/pipeline/operator/op_spec_test.cc

```diff
@@ -25,56 +25,50 @@
 
 namespace dali {
 
-class TestArgumentInput_Producer : public Operator<SupportBackend> {
+class TestArgumentInput_Producer : public Operator<CPUBackend> {
  public:
-  explicit TestArgumentInput_Producer(const OpSpec &spec) : Operator<SupportBackend>(spec) {}
+  explicit TestArgumentInput_Producer(const OpSpec &spec) : Operator<CPUBackend>(spec) {}
 
   bool CanInferOutputs() const override {
     return true;
   }
 
-  bool SetupImpl(std::vector<OutputDesc> &output_desc, const SupportWorkspace &ws) override {
-    output_desc.resize(4);
-    output_desc[0] = {uniform_list_shape(batch_size_, {1}), TypeInfo::Create<int>()};
-    output_desc[1] = {TensorListShape<>{{10}}, TypeInfo::Create<float>()};
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const HostWorkspace &ws) override {
+    output_desc.resize(3);
+    output_desc[0] = {uniform_list_shape(batch_size_, {1}), TypeTable::GetTypeInfo(DALI_INT32)};
     // Non-matching shapes
-    output_desc[2] = {uniform_list_shape(batch_size_ + 5, {1}), TypeInfo::Create<int>()};
-    output_desc[3] = {uniform_list_shape(batch_size_, {1, 2}), TypeInfo::Create<int>()};
+    output_desc[1] = {uniform_list_shape(batch_size_, {1}), TypeTable::GetTypeInfo(DALI_FLOAT)};
+    output_desc[2] = {uniform_list_shape(batch_size_, {1, 2}), TypeTable::GetTypeInfo(DALI_INT32)};
     return true;
   }
 
-  void RunImpl(SupportWorkspace &ws) override {
+  void RunImpl(HostWorkspace &ws) override {
     // Initialize all the data with a 0, 1, 2 .... sequence
     auto &out0 = ws.OutputRef<CPUBackend>(0);
     for (int i = 0; i < out0.shape().num_samples(); i++) {
-      *out0.mutable_tensor<int>(i) = i;
+      *out0[i].mutable_data<int>() = i;
     }
 
     auto &out1 = ws.OutputRef<CPUBackend>(1);
-    for (int i = 0; i < out1.shape()[0][0]; i++) {
-      out1.mutable_data<float>()[i] = i;
+    for (int i = 0; i < out1.shape().num_samples(); i++) {
+      *out1[i].mutable_data<float>() = i;
     }
 
     auto &out2 = ws.OutputRef<CPUBackend>(2);
     for (int i = 0; i < out2.shape().num_samples(); i++) {
-      *out2.mutable_tensor<int>(i) = i;
-    }
-
-    auto &out3 = ws.OutputRef<CPUBackend>(3);
-    for (int i = 0; i < out3.shape().num_samples(); i++) {
       for (int j = 0; j < 2; j++) {
-        out3.mutable_tensor<int>(i)[j] = i;
+        out2[i].mutable_data<int>()[j] = i;
       }
     }
   }
```

Comment:
you removed out3. Is it in purpose?

Reply:
This tested passing arguments of batch_size different than the one used in the whole pipeline and it's not possible now.